### PR TITLE
Feature/read all ft category implementation

### PR DESCRIPTION
**Pull request by MackowiakPiotr**

### CODE REVIEW

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/mapper/FinancialTransactionCategoryModelMapper.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package pl.byczazagroda.trackexpensesappbackend.mapper;
+
+import org.mapstruct.Mapper;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryDTO;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionCategory;
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/mapper/FinancialTransactionCategoryModelMapper.java:5]: Insert one blank line below, please.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/repository/FinancialTransactionCategoryRepository.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package pl.byczazagroda.trackexpensesappbackend.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionCategory;
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/repository/FinancialTransactionCategoryRepository.java:5]: Insert one blank line below, please.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionCategoryRepository;
+
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+@Validated
+public class FinancialTransactionCategoryServiceImpl implements FinancialTransactionCategoryService {
+
+    private final FinancialTransactionCategoryRepository financialTransactionCategoryRepository;
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java:17]: Insert one blank line below, please.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(controllers = FinancialTransactionCategoryController.class,
+        excludeFilters = @ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE,
+                classes = FinancialTransactionCategoryServiceImpl.class),
+        includeFilters = @ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE, classes = {FinancialTransactionCategoryModelMapper.class, ErrorStrategy.class}))
+@ActiveProfiles("test")
+class FinancialTransactionCategoryControllerTest {
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java:33]: Insert one blank line below, please.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    @Autowired
+    private MockMvc mockMvc;
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java:35]: Insert one blank line below, please.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    @MockBean
+    private FinancialTransactionCategoryService financialTransactionCategoryService;
+
+    @Test
+    @DisplayName("when finding all financial transaction categories " +
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java:40]: Maybe this display name will be better:
`when getting all financial transaction categories should return the financial transaction category DTOs list and response status OK`

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+    }
+
+    private List<FinancialTransactionCategoryDTO> createListOfFinancialTransactionCategoryDTOs() {
+        FinancialTransactionCategoryDTO categoryFirstDTO = new FinancialTransactionCategoryDTO(1L, "First", FinancialTransactionType.INCOME);
+        FinancialTransactionCategoryDTO categorySecondDTO = new FinancialTransactionCategoryDTO(2L, "Second", FinancialTransactionType.INCOME);
+        FinancialTransactionCategoryDTO categoryThirdDTO = new FinancialTransactionCategoryDTO(3L, "Third", FinancialTransactionType.INCOME);
+
+        return List.of(categoryFirstDTO, categorySecondDTO, categoryThirdDTO);
+    }
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java:66]: Insert one blank line below, please.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+        // then
+        mockMvc.perform(get("/api/categories")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.size()").value(3));
+
+    }
+
+    private List<FinancialTransactionCategoryDTO> createListOfFinancialTransactionCategoryDTOs() {
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java:60]: Maybe this method can be private and static.

Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java:60]: Method name "create Financial Transaction Category DTO List":
`createFinancialTransactionCategoryDTOList()`

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+        when(financialTransactionCategoryModelMapper.mapFinancialTransactionCategoryEntityToFinancialTransactionCategoryDTO(categoryThird))
+                .thenReturn(categoryThirdDTO);
+
+        List<FinancialTransactionCategoryDTO> returnedFinancialTransactionCategoryDTOsList = financialTransactionCategoryService.getFinancialTransactionCategories();
+
+        //then
+        Assertions.assertEquals(returnedFinancialTransactionCategoryDTOsList.get(0), categoryFirstDTO);
+        Assertions.assertEquals(returnedFinancialTransactionCategoryDTOsList.get(1), categorySecondDTO);
+        Assertions.assertEquals(returnedFinancialTransactionCategoryDTOsList.get(2), categoryThirdDTO);
+    }
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImplTest.java:63]: Insert one blank line below, please.

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Changes required

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/mapper/FinancialTransactionCategoryModelMapper.java (hunk 1 above)
Comment (MackowiakPiotr) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/mapper/FinancialTransactionCategoryModelMapper.java:5]: done

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/repository/FinancialTransactionCategoryRepository.java (hunk 2 above)
Comment (MackowiakPiotr) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/repository/FinancialTransactionCategoryRepository.java:5]: done

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java (hunk 4 above)
Comment (MackowiakPiotr) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java:33]: done

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java (hunk 5 above)
Comment (MackowiakPiotr) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java:36]: done

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java (hunk 6 above)
Comment (MackowiakPiotr) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java]: done

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    @MockBean
+    private FinancialTransactionCategoryService financialTransactionCategoryService;
+
+    @Test
+    @DisplayName("when finding all financial transaction categories " +
+            "should return financial transaction category DTOs list and response status OK")
+    void shouldResponseStatusOKAndFinancialTransactionCategoryDTOsList() throws Exception {
+
+        // given
+        List<FinancialTransactionCategoryDTO> listDTOs = createListOfFinancialTransactionCategoryDTOs();
```
[the remark this answers is not in the document]
Comment (MackowiakPiotr) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java]: done

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java (hunk 8 above)
Comment (MackowiakPiotr) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java]: done

Comment (MackowiakPiotr) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java]: done

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java (hunk 7 above)
Comment (MackowiakPiotr) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java:68]: done

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImplTest.java (hunk 9 above)
Comment (MackowiakPiotr) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImplTest.java:63]: done

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    @MockBean
+    private FinancialTransactionCategoryService financialTransactionCategoryService;
+
+    @Test
+    @DisplayName("when getting all financial transaction categories " +
+            "should return financial transaction category DTOs list and response status OK")
+    void shouldResponseStatusOKAndFinancialTransactionCategoryDTOsList() throws Exception {
+
+        // given
+        List<FinancialTransactionCategoryDTO> categoriesListDTO = createListOfFinancialTransactionCategoryDTOs();
```
Comment (odzio33) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryControllerTest.java:47]: What is the method createListOfFinancialTransactionCategoryDTOs? - Is not implemented. 
Please check the names of methods and variables in this file

**Review by odzio33 — CHANGES_REQUESTED** [review]
Changes required

**Review by odzio33 — APPROVED** [review]
Looks good for me

**Review by Krisu93 — APPROVED** [review]

**Review by odzio33 — APPROVED** [review]
Looks good for me

**Review by dawciobiel — APPROVED** [review]
Looks for me.

**Pull request merged**